### PR TITLE
AMDGPU: Document more backend recognized attributes

### PR DESCRIPTION
**Pull request by arsenm**

### CODE REVIEW

File: llvm/docs/AMDGPUUsage.rst [hunk 1]
```diff
@@ -1312,6 +1312,31 @@ The AMDGPU backend supports the following LLVM IR attributes.
...
+
+    "amdgpu-32bit-address-high-bits"         Assumed high 32-bits for
+                                             32-bit address spaces which are really truncated
+                                             64-bit addresses (i.e., addrspace(6))
+
+    "amdgpu-color-export"                    Assumed 1 for :ref:`amdgpu_ps <amdgpu_ps>`, and 0 for other calling conventions.
+
+    "amdgpu-depth-export"                    ..TODO:: Describe.
+
+    "InitialPSInputAddr"                     ..TODO:: Describe.
```
Comment (perlfu) [inline llvm/docs/AMDGPUUsage.rst:1329]:
```
Set the initial value of the `spi_ps_input_addr` register for :ref:`amdgpu_ps <amdgpu_ps>` shaders.
Any bits enabled by this value will be enabled in the final register value.
```

File: llvm/docs/AMDGPUUsage.rst [hunk 2]
```diff
@@ -1312,6 +1312,31 @@ The AMDGPU backend supports the following LLVM IR attributes.
...
+
+
+    "amdgpu-wave-priority-threshold"         ..TODO:: Describe.
+
+    "amdgpu-memory-bound".                   Set internally by backend
+
+    "amdgpu-wave-limiter"                    Set internally by backend
+
+    "amdgpu-unroll-threshold"                ..TODO:: Describe.
```
Comment (perlfu) [inline llvm/docs/AMDGPUUsage.rst:1338]:
```
Set base cost threshold preference for loop unrolling within this function, default is 300.
Actual threshold may be varied by per-loop metadata or reduced by heuristics.
```

File: llvm/docs/AMDGPUUsage.rst [hunk 3]
```diff
@@ -1438,7 +1438,10 @@ The AMDGPU backend supports the following LLVM IR attributes.
                                              :ref:`amdgpu_ps <amdgpu_ps>` shaders. Any bits enabled by this value will
                                              be enabled in the final register value.
 
-    "amdgpu-wave-priority-threshold"         ..TODO:: Describe.
+    "amdgpu-wave-priority-threshold"         VALU instruction count threshold for adjusting wave priority. If exceeded,
```
Comment (arsenm) [inline llvm/docs/AMDGPUUsage.rst:1441]: ping @kosarev

Comment (kosarev) [reply llvm/docs/AMDGPUUsage.rst]: The description looks correct to me. The pass was never proven useful though, if I recall correctly.

**Comment by arsenm** [comment]
ping

**Comment by arsenm** [comment]
ping

**Review by jayfoad — APPROVED** [review]

File: llvm/docs/AMDGPUUsage.rst [hunk 4]
```diff
@@ -1449,6 +1449,42 @@ The AMDGPU backend supports the following LLVM IR attributes.
...
+                                             be enabled in the final register value.
+
+     "amdgpu-wave-priority-threshold"        VALU instruction count threshold for adjusting wave priority. If exceeded,
+                                             temporarily raise the wave priority at the start of the shader function
+                                             until its last VMEM instructions to allow younger waves to issue their VMEM
+                                             instructions as well.
+
+     "amdgpu-memory-bound"                   Set internally by backend
+
+     "amdgpu-wave-limiter"                   Set internally by backend
```
Comment (jayfoad) [inline llvm/docs/AMDGPUUsage.rst:1482]: [LLPC](https://github.com/GPUOpen-Drivers/llpc) sets these in some cases. Any explicit setting takes precedence over the backend's internal heuristics. But I'm not sure if we want to document that.

**Pull request merged**